components/core/src/clp_s/search/QueryRunner.hpp

```diff
@@ -0,0 +1,438 @@
+#ifndef CLP_S_SEARCH_QUERYRUNNER_HPP
+#define CLP_S_SEARCH_QUERYRUNNER_HPP
+
+#include <cstddef>
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <optional>
+#include <set>
+#include <stack>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include <simdjson.h>
+
+#include "../ArchiveReader.hpp"
+#include "../ColumnReader.hpp"
+#include "../DictionaryReader.hpp"
+#include "../ReaderUtils.hpp"
+#include "../SchemaReader.hpp"
+#include "../SchemaTree.hpp"
+#include "../TimestampDictionaryReader.hpp"
+#include "../Utils.hpp"
+#include "ast/ColumnDescriptor.hpp"
+#include "ast/Expression.hpp"
+#include "ast/FilterExpr.hpp"
+#include "ast/FilterOperation.hpp"
+#include "ast/Literal.hpp"
+#include "clp_search/Query.hpp"
+#include "SchemaMatch.hpp"
+
+using namespace simdjson;
+using namespace clp_s::search::clp_search;
+
+namespace clp_s::search {
+/**
+ * This class is a core component of the log search system responsible for executing parsed queries
+ * represented as abstract syntax trees (ASTs). It sets up the necessary context for each schema and
+ * evaluates filter expressions against log messages from various schemas. It optimizes query
+ * execution through techniques like constant propagation and pre-processing. It also leverages
+ * dictionary lookups for efficient handling of string-based queries and supports wildcard and
+ * array-based filtering.
+ */
+class QueryRunner : public FilterClass {
+public:
+    QueryRunner(
+            std::shared_ptr<SchemaMatch> const& match,
+            std::shared_ptr<ast::Expression> const& expr,
+            std::shared_ptr<ArchiveReader> const& archive_reader,
+            bool ignore_case
+    )
+            : m_archive_reader(archive_reader),
+              m_expr(expr),
+              m_match(match),
+              m_ignore_case(ignore_case),
+              m_schema_tree(m_archive_reader->get_schema_tree()),
+              m_var_dict(m_archive_reader->get_variable_dictionary()),
+              m_log_dict(m_archive_reader->get_log_type_dictionary()),
+              m_array_dict(m_archive_reader->get_array_dictionary()),
+              m_timestamp_dict(m_archive_reader->get_timestamp_dictionary()),
+              m_schemas(m_archive_reader->get_schema_map()) {}
+
+    // Destructor
+    virtual ~QueryRunner() = default;
+
+    QueryRunner(QueryRunner const&) = delete;
+    auto operator=(QueryRunner const&) -> QueryRunner& = delete;
+
+    QueryRunner(QueryRunner&&) = delete;
+    auto operator=(QueryRunner&&) -> QueryRunner& = delete;
+
+    /**
+     * Initializes the query processing context for the first time.
+     */
+    void global_init();
+
+    /**
+     * Prepares the query processing context for a given schema.
```

Comment:
```suggestion
     * Initializes query processing context for a given schema.
```
Just to be a bit more consistent with how we word the docstring for `global_init`.